src/mapcss/Rule.py

```diff
@@ -115,7 +115,7 @@ def _test_feature_compatibility(f1, f2):
         return True
 #      else:
 #        return False
-    elif f2 == "line" and f1 in ("way", "line", "area"):
```

Comment:
need to change that since compatibility between line and area resulted in duplication:
polygons layer in mvt = polygons + lines
lines layer in mvt = lines

Reply:
it should be backwards - mvt line layer should have both polygons and lines, polygons layer can't have lines as there are no fills.

in OSM everything except node is a line.

Reply:
yes, I agree that I am trying to solve the problem in a wrong place. It seems like the actual problem is that osm2pgsql with default config imports boundaries in both tables planet_osm_polygon and planet_osm_line and they are processed and included in tiles twice.

Reply:
putting both lines and polygons in the same mvt layer makes sense for me. moreover it is possible to put everything in single mvt layer but my main concern is that mapbox suggests to have less filtering rules in styles https://docs.mapbox.com/help/troubleshooting/mapbox-gl-js-performance/#reduce-the-number-of-filters while it won't be possible when everything is stored in one layer. we don't have performance problem at the moment but I don't know what is going to happen for complicated mapcss.